timeboost/auctioneer.go

```diff
@@ -263,16 +275,48 @@ func (a *AuctioneerServer) Start(ctx_in context.Context) {
 				// There's nothing in the queue.
 				return time.Millisecond * 250
 			}
+
+			if err := validateBidTemporal(&a.roundTimingInfo, (uint64)(req.Value.Round)); err != nil {
```

Comment:
It would have been nicer to have the body of this CallIteratively invocation be its own function, which we can more easily unit test. It seems there are several assertions we would want to ensure in this logic, for instance, that we do not send duplicate bids to the receiver because it cares about orderings. We can also check that we reconsumed the same bid, etc. This business logic could be tested by passing in a mock redis interface as well

Reply:
Moved it to its own function consumeNextBid and added unit tests.